homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,317 @@
+    SIGNAL_ADD_X10_DEVICE,
+    SIGNAL_REMOVE_DEVICE_OVERRIDE,
+    SIGNAL_REMOVE_X10_DEVICE,
+)
+from .schemas import (
+    add_device_override,
+    add_x10_device,
+    build_device_override_schema,
+    build_hub_schema,
+    build_plm_schema,
+    build_remove_override_schema,
+    build_remove_x10_schema,
+    build_x10_schema,
+)
+
+STEP_PLM = "plm"
+STEP_HUB_V1 = "hubv1"
+STEP_HUB_V2 = "hubv2"
+STEP_CHANGE_HUB_CONFIG = "change_hub_config"
+STEP_ADD_X10 = "add_x10"
+STEP_ADD_OVERRIDE = "add_override"
+STEP_REMOVE_OVERRIDE = "remove_override"
+STEP_REMOVE_X10 = "remove_x10"
+MODEM_TYPE = "modem_type"
+PLM = "PowerLinc Modem (PLM)"
+HUB1 = "Hub version 1 (pre-2014)"
+HUB2 = "Hub version 2"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _only_one_selected(*args):
+    """Test if only one item is True."""
+    return sum(args) == 1
+
+
+async def _async_connect(**kwargs):
+    """Connect to the Insteon modem."""
+    try:
+        await async_connect(**kwargs)
+        _LOGGER.info("Connected to Insteon modem.")
+        return True
+    except ConnectionError:
+        _LOGGER.error("Could not connect to Insteon modem.")
+        return False
+
+
+def _remove_override(address, options):
+    """Remove a device override from config."""
+    new_options = {}
+    if options.get(CONF_X10):
+        new_options[CONF_X10] = options.get(CONF_X10)
+    new_overrides = []
+    for override in options[CONF_OVERRIDE]:
+        if override[CONF_ADDRESS] != address:
+            new_overrides.append(override)
+    if new_overrides:
+        new_options[CONF_OVERRIDE] = new_overrides
+    return new_options
+
+
+def _remove_x10(device, options):
+    """Remove an X10 device from the config."""
+    housecode = device[11].lower()
+    unitcode = int(device[24:])
+    new_options = {}
+    if options.get(CONF_OVERRIDE):
+        new_options[CONF_OVERRIDE] = options.get(CONF_OVERRIDE)
+    new_x10 = []
+    for existing_device in options[CONF_X10]:
+        if (
+            existing_device[CONF_HOUSECODE].lower() != housecode
+            or existing_device[CONF_UNITCODE] != unitcode
+        ):
+            new_x10.append(existing_device)
+    if new_x10:
+        new_options[CONF_X10] = new_x10
+    return new_options, housecode, unitcode
+
+
+class InsteonFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Insteon config flow handler."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Define the config flow to handle options."""
+        return InsteonOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """For backward compatibility."""
+        return await self.async_step_init(user_input=user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Init the config flow."""
+        errors = {}
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
```

Comment:
If only one entry is allowed we use the `single_instance_allowed` abort reason.

https://github.com/home-assistant/core/blob/f286992b1007588ae1221352883259393947fdfe/homeassistant/strings.json#L50